src/MimeUploadValidator.php

```diff
@@ -139,7 +139,8 @@ public function getExpectedMimeTypes($file)
      */
     public function compareMime($first, $second)
     {
-        return preg_replace($this->filterPattern, '', $first) === preg_replace($this->filterPattern, '', $second);
+        $b = preg_replace($this->filterPattern ?? '', '', $second ?? '');
+        return preg_replace($this->filterPattern ?? '', '', $first ?? '') === $b;
```

Comment:
```suggestion
        $b = preg_replace($this->filterPattern ?? '//', '', $second ?? '');
        return preg_replace($this->filterPattern ?? '//', '', $first ?? '') === $b;
```
Passing an empty string instead of an empty regex results in a different warning being emitted.

Reply:
Not clear what the reasoning was for assigning the `$b` variable like that but it doesn't really affect anything so I'm fine with it. I'm _guessing_ it was a phpcs issue with too many characters on one line, maybe?

Reply:
As per comments on another similar PR - the warning that gets emitted for the empty string is correct and points at the issue which already exists with the null value. This is only about resolving the deprecation warning emitted by passing null.